Source/lighting.cpp

```diff
@@ -509,11 +509,11 @@ void DoLighting(Point position, int nRadius, int Lnum)
 		yoff = LightList[Lnum].position.offset.y;
 		if (xoff < 0) {
 			xoff += 8;
-			position -= { 1, 0 };
+			position -= Size { 1, 0 };

```

Comment:
Can you elaborate why using directions is more appropriate here?

Reply:
This is an odd bit of code, but effectively it's used to offset the actual position of a light which occupies multiple tiles.

It would be clearer as a set of statements with three cases:
```cpp
if (yoff >= 0 && xoff < 0) {
    // handle a light offset to the top left relative to the screen: [ \ ]
    position += DIR_NW;
} else if (yoff < 0 && xoff < 0) {
    // handle a light offset to the top relative to the screen: [ | ]
    position += DIR_N;
} else if (yoff < 0 && xoff >= 0) {
    // handle a light offset to the top right relative to the screen: [ / ]
    position += DIR_NE
} // else no offset needed
```

The current code shortcuts the middle case by taking advantage of `{-1, 0} + {0, -1} == {-1, -1}`